src/yunohost/user.py

```diff
@@ -324,7 +324,9 @@ def user_update(operation_logger, username, firstname=None, lastname=None, mail=
     if lastname and firstname:
         new_attr_dict['cn'] = new_attr_dict['displayName'] = [firstname + ' ' + lastname]
 
-    if change_password:
+    if change_password is not None:
+        if not change_password:
```

Comment:
The moulinette framework generate an HTTP API and a CLI with the same code.

You should add a test to check you are in the CLI context.

```suggestion
        if not change_password:
```
        if not change_password and msettings.get('interface') == 'cli':
```

Reply:
Oh right, I didn't realize that was the case, I added this. thanks!